src/coreclr/nativeaot/System.Private.CoreLib/src/System/Runtime/InteropServices/ComWrappers.CoreRT.cs

```diff
@@ -0,0 +1,452 @@
+        /// </summary>
+        private static ComWrappers? s_globalInstanceForTrackerSupport;
+#endif
+
+        /// <summary>
+        /// Globally registered instance of the ComWrappers class for marshalling.
+        /// </summary>
+        private static ComWrappers? s_globalInstanceForMarshalling;
+
+        /// <summary>
+        /// Create a COM representation of the supplied object that can be passed to a non-managed environment.
+        /// </summary>
+        /// <param name="instance">The managed object to expose outside the .NET runtime.</param>
+        /// <param name="flags">Flags used to configure the generated interface.</param>
+        /// <returns>The generated COM interface that can be passed outside the .NET runtime.</returns>
+        /// <remarks>
+        /// If a COM representation was previously created for the specified <paramref name="instance" /> using
+        /// this <see cref="ComWrappers" /> instance, the previously created COM interface will be returned.
+        /// If not, a new one will be created.
+        /// </remarks>
+        public unsafe IntPtr GetOrCreateComInterfaceForObject(object instance, CreateComInterfaceFlags flags)
+        {
+            if (instance == null)
+                throw new ArgumentNullException(nameof(instance));
+
+            ManagedObjectWrapperHolder ccwValue;
+            if (_ccwTable.TryGetValue(instance, out ccwValue))
+            {
+                return ccwValue.ComIp;
+            }
+
+            ccwValue = _ccwTable.GetValue(instance, (c) =>
+            {
+                ManagedObjectWrapper* value = CreateCCW(this, c, flags);
+                return new ManagedObjectWrapperHolder(value);
+            });
+            return ccwValue.ComIp;
+        }
+
+        private static unsafe ManagedObjectWrapper* CreateCCW(ComWrappers impl, object instance, CreateComInterfaceFlags flags)
+        {
+            ComInterfaceEntry* userDefined = impl.ComputeVtables(instance, flags, out int userDefinedCount);
+
+            // Maximum number of runtime supplied vtables.
+            Span<IntPtr> runtimeDefinedVtable = stackalloc IntPtr[4];
+            int runtimeDefinedCount = 0;
+
+            // Check if the caller will provide the IUnknown table.
+            if ((flags & CreateComInterfaceFlags.CallerDefinedIUnknown) == CreateComInterfaceFlags.None)
+            {
+                runtimeDefinedVtable[runtimeDefinedCount++] = DefaultIUnknownVftblPtr;
+            }
+
+            // Compute size for ManagedObjectWrapper instance.
+            int totalDefinedCount = runtimeDefinedCount + userDefinedCount;
+
+            // Compute the total entry size of dispatch section.
+            int totalDispatchSectionSize = totalDefinedCount * sizeof(InternalComInterfaceDispatch);
+
+            // Allocate memory for the ManagedObjectWrapper.
+            IntPtr wrapperMem = Marshal.AllocCoTaskMem(
+                sizeof(ManagedObjectWrapper) + totalDispatchSectionSize);
```

Comment:
```suggestion
            // Allocate memory for the ManagedObjectWrapper.
            IntPtr wrapperMem = Marshal.AllocCoTaskMem(
                sizeof(ManagedObjectWrapper) + totalDefinedCount * sizeof(InternalComInterfaceDispatch));
```
Nit. Writing it this way makes is easier to see what's stored in the memory block.